UOP1_Project/Assets/Scripts/SpawnSystem.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Linq;
+using UnityEditor.PackageManager;
+using UnityEngine;
+using UnityEngine.Events;
+
+[Serializable]
+public class PlayerSpawnEvent : UnityEvent<Transform> { }
```

Comment:
New to C# and Unity, why is this empty class here? Is there a better way to inherit from UnityEvent<Transform> ?

Reply:
It's done so it can be serialized and show up in the inspector.
Doing `public UnityEvent<Transform> onPlayerSpawnEvent;` would work the same way at C# level but can't be manipulated by the level designer in the editor.

Reply:
FWIW, it's no longer necessary in 2020.1, but this project is using 2019 LTS so indeed it's still necessary for now.